### PR TITLE
[#14820] Size method with a local cache and store is not optimized

### LINKED ISSUE #14820: Size method with a local cache and store is not optimized
### Describe the bug

The size optimization in CacheLoaderInterceptor has a check for if a command contains CACHE_MODE_LOCAL but it is missing a check if the cache itself is already local. This means it requires a full iteration to obtain the size instead of using an optimized size check.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Infinispan version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by wburns**
Fixes #14820

### CODE REVIEW

File: core/src/main/java/org/infinispan/interceptors/impl/CacheLoaderInterceptor.java [hunk 1]
```diff
@@ -278,7 +283,7 @@ private CompletionStage<Long> trySizeOptimization(SizeCommand command) {
 
       // Get the size from any shared store that isn't async
       return persistenceManager.size(SHARED.and(NOT_ASYNC))
-            .thenCompose(v -> v >= 0 || !command.hasAnyFlag(FlagBitSets.CACHE_MODE_LOCAL)
+            .thenCompose(v -> v >= 0 || (!command.hasAnyFlag(FlagBitSets.CACHE_MODE_LOCAL) && !isLocalCache)
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/interceptors/impl/CacheLoaderInterceptor.java:286]: I suggest to take the second condition out of the lambda to avoid creating it. 

```java
      var stage = persistenceManager.size(SHARED.and(NOT_ASYNC));
      if (!command.hasAnyFlag(FlagBitSets.CACHE_MODE_LOCAL) && !isLocalCache) {
         return stage;
      }
      return stage.thenCompose(v -> v >= 0 ? ...)
```

**Review by pruivo — APPROVED** [review]
I just wrote a small suggestion. The PR looks good to me anyway! :+1:

**Review by pruivo — APPROVED** [review]

**Comment by wburns** [comment]
I am not sure why CI won't complete with this change though.. it has failed for two different reasons already which is unexpected.

**Review by jabolina — APPROVED** [review]

**Comment by jabolina** [comment]
Seems the failures are related.

**Comment by wburns** [comment]
Should be fixed now. Two were just test related and the other was a weird edge case with SIFS if you deleted an index file but left others, which is extremely rare to happen as you would have to force kill the JVM between writing two very small files.

**Comment by pruivo** [comment]
merged! thanks @wburns !
